test/dynamo/test_misc.py

```diff
@@ -2954,6 +2954,27 @@ def forward(self, x):
         res = opt_model(x)
         self.assertTrue(same(ref, res))
 
+    def test_if_cond_user_defined_object(self):
+        class MyClass(object):
+            pass
+
+        def fn(x, obj):
+            if not obj:
```

Comment:
For this pattern, I think you can check if the class has `__bool__` and if not return map it to a constant `True`.